common/djangoapps/student/models.py

```diff
@@ -3086,6 +3091,67 @@ def should_celebrate_first_section(enrollment):
             return False
 
 
+class UserCelebration(TimeStampedModel):
+    """
+    Keeps track of how we've celebrated a user's progress.
+
+    An example of a celebration is a dialog that pops up after you post for the first time in the
+    discussion forums.
+
+    In general, these celebrations should be course agnostic and could show up at any point for
+    a learner. Some will be one-off celebrations (like first discussion post) while others could
+    be repeated for a user (like streaks).
+    For celebrations that can be repeated with each enrollment, see CourseEnrollmentCelebration
+
+    .. no_pii:
+    """
+    user = models.OneToOneField(User, models.CASCADE, related_name='celebrations')
+    # Use a NullBooleanField in case another celebration creates the row.
+    celebrate_first_discussion_post = NullBooleanField(default=None)
+
+    def __str__(self):
+        return (
+            "[UserCelebration] user: {}; first_discussion_post: {}"
+        ).format(self.user.username, self.celebrate_first_discussion_post)
+
+    @staticmethod
+    def should_celebrate_first_discussion(user, course_key):
+        """
+        Returns the celebration value for first_discussion.
+
+        If the model does not exist or the value is None (i.e. we don't yet know if the learner has
+        posted their first discussion), we will send a request to the forums service to see if they
+        have ever posted.
+        """
+        # if not user or user.is_anonymous or not courseware_mfe_first_discussion_celebration_is_active(course_key):
+        if not user or user.is_anonymous:
+            return False
+        try:
+            should_celebrate = user.celebrations.celebrate_first_discussion_post
+            # should_celebrate will be None if a different celebration type created the row for this
+            # user. In that case, we will query the forums service (outside of try-except) to find
+            # out what the value should be and fill it in.
+            if should_celebrate is not None:
+                return should_celebrate
+        except UserCelebration.DoesNotExist:
+            # If UserCelebration does not exist for this user, we will create it below with the
+            # correct value from the forums service
+            pass
+
+        # ask forums and create or update row with value
+        url = forums_api_prefix + '/users/{}/content_exists'.format(user.id)
+        # Returns a bool
+        try:
+            has_posted_before = perform_forums_request('get', url)
```

Comment:
This is a synchronous request to a service, backend-to-backend. That's not our preferred model. We like MFEs to ask backends, but not for them to sync talk to each other, iirc. Is there a way to have the MFE talk to the forums on this? Sorry that this foundational question is being raised so late. Or is this flow temporary anyway, until the discussion MFE is up and running?

Also, I know you mention that we might send a request in the docstring, but you might triple underline that this method blocks for an indeterminate amount of time, and shouldn't be used willy nilly without being ready for that.